modules/costs/spec/features/timer_spec.rb

```diff
@@ -0,0 +1,157 @@
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require_relative '../spec_helper'
+
+RSpec.describe 'Work Package timer', js: true do
+  shared_let(:project) { create(:project_with_types) }
+
+  shared_let(:work_package_a) { create(:work_package, subject: 'WP A', project:) }
+  shared_let(:work_package_b) { create(:work_package, subject: 'WP B', project:) }
+
+  let(:wp_view_a) { Pages::FullWorkPackage.new(work_package_a) }
+  let(:wp_view_b) { Pages::FullWorkPackage.new(work_package_b) }
+  let(:time_logging_modal) { Components::TimeLoggingModal.new }
+  let(:timer_button) { Components::WorkPackages::TimerButton.new }
+
+  let(:user) { create(:user, member_in_project: project, member_with_permissions: permissions) }
+
+  before do
+    login_as user
+  end
+
+  shared_examples 'allows time tracking' do
+    it 'shows the timer and allows tracking time' do
+      wp_view_a.visit!
+      timer_button.expect_visible
+      timer_button.start
+      timer_button.expect_active
+
+      active_time_entries = TimeEntry.where(ongoing: true, user:)
+      expect(active_time_entries.count).to eq 1
+      timer_entry = active_time_entries.first
+      expect(timer_entry.work_package).to eq work_package_a
+      expect(timer_entry.hours).to be_nil
+
+      page.find('.op-top-menu-user').click
+      expect(page).to have_selector('.op-timer-account-menu', wait: 10)
+      expect(page).to have_selector('.op-timer-account-menu--wp-details', text: "##{work_package_a.id}: WP A")
+      page.find('[data-qa-selector="op-timer-account-menu-stop"]').click
+
+      time_logging_modal.is_visible true
+
+      time_logging_modal.has_field_with_value 'spentOn', Date.current.strftime
+      time_logging_modal.has_field_with_value 'hours', /(\d\.)?\d+/
+      time_logging_modal.work_package_is_missing false
+
+      time_logging_modal.perform_action 'Save'
+      time_logging_modal.is_visible false
+
+      wp_view_a.expect_and_dismiss_toaster message: I18n.t(:notice_successful_update)
+      timer_button.expect_inactive
+
```

Comment:
The specs are passing successfully, however when I run the specs in chrome, I'll get the following error message toast right after the success toast:

<img width="1279" alt="image" src="https://github.com/opf/openproject/assets/83396/d11b1730-cd75-4591-8244-544b638c8c7c">